### ISSUE TITLE
Named windows don't work inside WITH clauses

### ISSUE THREAD

**Issue by georgewfraser** [cwida/duckdb#546]
Consider this simple window function query:

```sql
select i, lag(i) over named_window
from (values (1), (2), (3)) as t (i)
window named_window as (order by i);
```

which produces this output:

```
i           WINDOW    
----------  ----------
1                     
2           1         
3           2   
```

Now suppose I put it inside a `with` clause:

```sql
with subquery as (
    select i, lag(i) over named_window
    from (values (1), (2), (3)) as t (i)
    window named_window as (order by i)
)
select * from subquery;
```

Now it produces an error:

```
Error: Could not find named window specification
```

Bizarrely, if you move the window frame to the parent, it works:

```sql
with subquery as (
    select i, lag(i) over named_window
    from (values (1), (2), (3)) as t (i)
)
select * from subquery
window named_window as (order by i);
```

I'm pretty sure this is a bug.

**Comment by Mytherin** [comment]
Indeed, this looks like a bug! Thanks for the report. At a first glance it seems like named windows are query wide, rather than restricted to the select node as they should be.

**Comment by Mytherin** [comment]
Should be fixed in 00a2946aaf972670dd3bfca750b1ae706d2fd530, will merge into master after the tests pass.
